### PR TITLE
Set compression to zstd, and support flow ignore kafka opts parameters

### PR DESCRIPTION
**Pull request by galuhest**

### CODE REVIEW

File: cmds/action.go [hunk 1]
```diff
@@ -107,11 +109,15 @@ func ActionBaritoProducerService(c *cli.Context) (err error) {
 
 	// kafka producer config
 	config := sarama.NewConfig()
-	config.Producer.RequiredAcks = sarama.WaitForAll
+	config.Producer.RequiredAcks = 1
```
Comment (ekarisky) [inline cmds/action.go:112]: do we need to change this? this mean only leaders will response, not all member

File: cmds/action.go [hunk 2]
```diff
@@ -107,11 +109,15 @@ func ActionBaritoProducerService(c *cli.Context) (err error) {
...
 	config.Producer.Retry.Max = maxRetry
 	config.Producer.Return.Successes = true
+	config.Producer.Compression = sarama.CompressionZSTD
+	config.Producer.CompressionLevel = sarama.CompressionLevelDefault
 	config.Metadata.RefreshFrequency = 1 * time.Minute
-	config.Version = sarama.V0_10_2_1 // TODO: get version from env
+	config.Producer.Flush.Bytes = 16000
```
Comment (ekarisky) [inline cmds/action.go:118]: whats effect after this changes?

File: flow/grpc_producer.go [hunk 3]
```diff
@@ -270,8 +272,13 @@ func (s *producerService) sendCreateTopicEvents(topic string) (err error) {
 
 func (s *producerService) handleProduce(timber *pb.Timber, topic string) (err error) {
 	if !s.admin.Exist(topic) {
-		numPartitions := timber.GetContext().GetKafkaPartition()
-		replicationFactor := timber.GetContext().GetKafkaReplicationFactor()
+		var numPartitions int32 = -1
```
Comment (ekarisky) [inline flow/grpc_producer.go:275]: confirmed not `0`?

File: cmds/action.go (hunk 1 above)
Comment (bugbounce) [reply cmds/action.go:112]: This is to reduce the producer latency. We don't have enough reason to need all producer acks. Kafka rarely goes down and unless we have a SLA that is broken, we should keep this for performance.

File: cmds/action.go (hunk 2 above)
Comment (bugbounce) [reply cmds/action.go:118]: This works in combination with the flush frequency. 16Mb is the Kafka default.

File: flow/grpc_producer.go (hunk 3 above)
Comment (bugbounce) [reply flow/grpc_producer.go:275]: This is to use the default received from broker properties.

File: cmds/action.go (hunk 1 above)
Comment (ekarisky) [reply cmds/action.go:112]: okay, understood

**Review by ekarisky — APPROVED** [review]

**Pull request merged**